api/applications/serializers.py

```diff
@@ -8,31 +8,34 @@
 
 class ApiOAuth2ApplicationSerializer(JSONAPISerializer):
     """Serialize data about a registered OAuth2 application"""
-    client_id = ser.CharField(help_text="The client ID for this application (automatically generated)",
+    id = ser.CharField(help_text='The client ID for this application (automatically generated)',
+                       read_only=True,
```

Comment:
An `OPTIONS` request to a detail endpoint will return metadata, e.g. the expected parameters (which correspond to serializer). 

For example, if you send an OPTIONS request for the user detail URL for your own user, you get:


![django_rest_framework](https://cloud.githubusercontent.com/assets/2379650/9691818/ea2a4946-5312-11e5-8fe8-6e30ad75a111.png)